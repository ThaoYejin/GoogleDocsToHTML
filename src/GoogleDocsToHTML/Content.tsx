import { docs_v1 } from "googleapis";
import React from "react";
import { Heading } from "./Heading";
import { Paragraph } from "./Paragraph";
import { Table } from "./Table";
export interface ContentProps {
    doc: docs_v1.Schema$Document;
    body? : docs_v1.Schema$StructuralElement[];
}

export const Content: React.FC<ContentProps> = ({
    doc,
    body
}) => {
    var content: React.ReactNode[];
    content = [];
    body?.forEach(function (value, index) {
        var pObj = value.paragraph;

        //Display Heading
        if (pObj?.paragraphStyle?.namedStyleType?.indexOf("HEADING") != -1) {
            content?.push(
                <Heading key={index} paraObj={pObj}></Heading>
            )
        }

        //Display paragraph
        if (pObj?.paragraphStyle?.namedStyleType?.indexOf("HEADING") == -1 && value.table == undefined && pObj.bullet == undefined) {
            content?.push(<Paragraph key={index} paraObj={pObj} doc={doc}></Paragraph>)
        }

        //Display table
        if(value.table != undefined){
            content.push(<Table doc={doc} table={value.table}></Table>)
        }
    })
    return (
        <>
            {content}
        </>
    )
}