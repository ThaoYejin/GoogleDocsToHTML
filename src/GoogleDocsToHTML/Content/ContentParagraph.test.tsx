import React from 'react';
import renderer from 'react-test-renderer';
import { Content } from './Content';
import * as doc from './ContentParagraphTest.json';

it('renders correctly', () => {
  var body = [
    {
      "endIndex": 1,
      "sectionBreak": {
        "sectionStyle": {
          "columnSeparatorStyle": "NONE",
          "contentDirection": "LEFT_TO_RIGHT",
          "sectionType": "CONTINUOUS"
        }
      }
    },
    {
      "startIndex": 1,
      "endIndex": 34,
      "paragraph": {
        "elements": [
          {
            "startIndex": 1,
            "endIndex": 33,
            "textRun": {
              "content": "This is just a normal paragraph.",
              "textStyle": {
                "foregroundColor": {
                  "color": {
                    "rgbColor": {
                      "red": 0.2509804,
                      "green": 0.2509804,
                      "blue": 0.2509804
                    }
                  }
                },
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                }
              }
            }
          },
          {
            "startIndex": 33,
            "endIndex": 34,
            "textRun": {
              "content": "\n",
              "textStyle": {}
            }
          }
        ],
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT"
        }
      }
    }
  ];

    const tree = renderer
        .create(<Content doc={doc} body={body} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})