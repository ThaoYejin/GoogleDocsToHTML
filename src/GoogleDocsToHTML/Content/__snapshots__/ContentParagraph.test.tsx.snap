// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h6 />,
  <p>
    <span
      style={
        Object {
          "color": "rgb(64.000002,64.000002,64.000002)",
          "fontSize": 12,
        }
      }
    >
      This is just a normal paragraph.
    </span>
    <span
      style={Object {}}
    >
      

    </span>
  </p>,
]
`;
